sql/importer: [empty send-queue with non-zero precise size] TestImportIntoCSV failed

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17780119?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17780119?buildTab=artifacts#/) on master @ [d72000b025d709a8c7ec49e3004ad7befc1f2588](https://github.com/cockroachdb/cockroach/commits/d72000b025d709a8c7ec49e3004ad7befc1f2588):


```
*   | 	pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go:784
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	pkg/kv/kvserver/replica_raft.go:1017
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	pkg/kv/kvserver/replica_raft.go:836
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	pkg/kv/kvserver/store_raft.go:682
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	pkg/kv/kvserver/scheduler.go:419
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	pkg/kv/kvserver/scheduler.go:319
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_arm64.s:1222
* Wraps: (4) empty send-queue with non-zero precise size: before=+46 KiB after=+23 KiB delta=-23 KiB [queue_len=0, queue_approx_size=+23 KiB]
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
panic: empty send-queue with non-zero precise size: before=+46 KiB after=+23 KiB delta=-23 KiB [queue_len=0, queue_approx_size=+23 KiB] [recovered]
	panic: empty send-queue with non-zero precise size: before=+46 KiB after=+23 KiB delta=-23 KiB [queue_len=0, queue_approx_size=+23 KiB]

goroutine 7144 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400ae638e8?, {0x81868d0, 0x400177ef60})
	pkg/util/stop/stopper.go:226 +0x68
panic({0x6180fc0?, 0x4001af0c70?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/rac2.(*replicaSendStream).applySendQueuePreciseSizeDeltaRaftMuAndStreamLocked(0x0?, {0x81868d0?, 0x4008a5c510?}, 0xffffffffffffa3e3)
	pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go:2549 +0x158
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/rac2.(*replicaSendStream).handleReadyEntriesRaftMuAndStreamLocked(0x40080d4c40, {0x81868d0, 0x4008a5c510}, {0x0, {0x1, 0x14, 0x15}, 0x17, {0x401239d0e0, 0x0, ...}, ...}, ...)
	pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go:2631 +0x808
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/rac2.(*replicaState).handleReadyEntriesRaftMuLocked(0x400b0ac000, {0x81868d0, 0x4008a5c510}, {0x0, {0x1, 0x14, 0x15}, 0x17, {0x401239d0e0, 0x0, ...}, ...}, ...)
	pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go:2333 +0x134
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/rac2.(*rangeController).HandleRaftEventRaftMuLocked(0x400664a680, {0x81868d0, 0x4008a5c510}, {0x0, 0x0, 0x0, {0x0, 0x0, 0x0}, 0x4008a5c600, ...})
	pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go:1183 +0x8a4
github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/replica_rac2.(*processorImpl).HandleRaftReadyRaftMuLocked(0x400b77e008, {0x81868d0, 0x4008a5c510}, {0x6, 0x1, 0x1, 0x17, 0x1}, {0x0, 0x0, ...})
	pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go:784 +0x178
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	pkg/kv/kvserver/replica_raft.go:1017 +0x4b4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	pkg/kv/kvserver/replica_raft.go:836 +0x140
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0x4004771808, 0x81868d0?)
	pkg/kv/kvserver/store_raft.go:682 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0x4001382770, {0x81868d0, 0x400177ef60}, {0x81b6a00, 0x4004771808}, 0x400c710c88)
	pkg/kv/kvserver/scheduler.go:419 +0x240
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x81868d0?, 0x400177ef60?})
	pkg/kv/kvserver/scheduler.go:319 +0x4c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x6879b78?, 0x1244010041244010?})
	pkg/util/stop/stopper.go:498 +0x198
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 6967
	pkg/util/stop/stopper.go:488 +0x384
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129404 sql/importer: TestImportIntoCSV failed [C-test-failure O-robot P-3 T-sql-queries branch-release-24.1.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44438

--- a/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go
+++ b/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go
@@ -2601,6 +2601,7 @@ func (rss *replicaSendStream) handleReadyEntriesRaftMuAndStreamLocked(
 		}
 		rss.mu.sendQueue.indexToSend = event.sendingEntries[n-1].id.index + 1
 		var sendTokensToDeduct [admissionpb.NumWorkClasses]kvflowcontrol.Tokens
+		var sendQPreciseSizeDelta kvflowcontrol.Tokens
 		for _, entry := range event.sendingEntries {
 			if !entry.usesFlowControl {
 				continue
@@ -2628,10 +2629,13 @@ func (rss *replicaSendStream) handleReadyEntriesRaftMuAndStreamLocked(
 			if inSendQueue && entry.id.index >= rss.mu.nextRaftIndexInitial {
 				// Was in send-queue and had eval tokens deducted for it.
 				rss.mu.sendQueue.originalEvalTokens[WorkClassFromRaftPriority(entry.pri)] -= tokens
-				rss.applySendQueuePreciseSizeDeltaRaftMuAndStreamLocked(ctx, -tokens)
+				sendQPreciseSizeDelta -= tokens
 			}
 			rss.raftMu.tracker.Track(ctx, entry.id, pri, tokens)
 			sendTokensToDeduct[WorkClassFromRaftPriority(pri)] += tokens
+		}
+		if sendQPreciseSizeDelta != 0 {
+			rss.applySendQueuePreciseSizeDeltaRaftMuAndStreamLocked(ctx, sendQPreciseSizeDelta)
 		}
 		flag := AdjNormal
 		if directive.preventSendQNoForceFlush {

